### ISSUE TITLE
Revisit role of WarpCore

### ISSUE THREAD

**Issue by TheNumenorean** [LemADEC/WarpDrive#37]
In startrek, the warp core really is just a reactor, that outputs immense amounts of energy from antimatter and matter reactions. The Actual warping is accomplished by means of the Nacells, with energy from the warp core. We could remove the existing power gen options and change the warpcore to be the power option, making the Nacells then being the requirement for warp. This makes warpcores useful on non-ships, and means we could make them more interesting, possibly multiblock structures. It also means that the nacells could be required to be proportional to the ship size.

The other functions that the warp core currently has could also be given off to other blocks. If we get a transporter working, there is no need for the warpcore to be able to teleport players to it, which frankly doesn't make sense anyway.

Everything could still be controlled by computer, but it would instead be controlling the nacells separately. Greater power requirements for biger ships would require bigger warp cores, and thus ships would have to allocate room for them. We could also still make it possible to power nacells with electricity, but it would be extravagantly expensive. True to canon.

**Comment by LemADEC** [comment]
WarpDrive isn't specifically for Startrek and doesn't intend to be canon.
That being said, the current WarpCore is an 'all in one' block that doesn't make much sense.
Regarding power, there's already an Enantiomorphic reactor. It needs a revamp and eventually a tiered system, but it's there.
Current WarpCore includes:
- Teleportation, see related ticket for Transporter.
- Beacon jump which is functional but too complicated to use. An anchor block would be better suited.
- Forced teleportation which is merely an experimental feature and very OP
- Jump which should be Entity based movements
- Warp which is the actual core of the mod

We need to implement those base blocks and a Mainframe before we can split the existing WarpCore to atomic functions.

**Comment by CaptainMegavolt** [comment]
I kinda like the idea of some kind of external warp field emitters though.
PS will any of the blocks in this ever get some kind of special 3D model?

**Comment by LemADEC** [comment]
We have no 3D modeler, until then, they'll remain plain blocks

**Comment by CaptainMegavolt** [comment]
Maybe I could make a few models. If I do, I'll send em in to you.

_--Supernova Industries Supreme Chairman Megavolt_

**Comment by LemADEC** [comment]
Most of those are already done or no longer applicable, closing.
